### DISCUSSION TITLE
what is `SubscribePayload.extensions` for?

### DISCUSSION THREAD

**Discussion by jjangga0214** [enisdenjo/graphql-ws#210] [Q&A] [answered]
I looked around source code, but couldn't find how `SubscribePayload.extensions` is practically used.

Is it something like `context`, but not implemented yet?

Thanks :)

**Comment by enisdenjo** [comment, marked as answer]
_*Converted to a Q&A discussion_

The `extensions` field is reserved for implementors to extend the protocol however they see fit. Through it you can pass extra values to the GraphQL operation for custom behaviour, one example being: [persisted queries](https://blog.codecentric.de/en/2020/05/how-to-secure-a-graphql-service-using-persisted-queries/).

**Reply by enisdenjo** [reply]
FYI, I've [changed](https://github.com/enisdenjo/graphql-ws/commit/f2a92aff2be9e700d1931f0c600c71b67146f617) the ["ws server and client usage with persisted queries" recipe](https://github.com/enisdenjo/graphql-ws#persisted) to showcase the convention of using the `extensions` field.
